### PR TITLE
fix: deep comparing props (fix #26)

### LINKED ISSUE #26: Handling calendar events with useState() hook
## Version
v1.0.5
## Test Environment
Browsers - Firefox, chrome
OS - Windows

## Current Behavior
My usecase requires me to update the local state of the application when user clicks on a tile. 

By the documentation, clicking on a tile fires the "beforeCreateSchedule" event, so naturally I've tried collecting it (as it contains everything I need) and saving it in the local state by using the setter function returned by useState.

But doing so leads to re-rendering the page, this causes the "guide" to disappear. It also seems to strip the event of some data. While, that stripped data isn't essential for me, the disappearance of the guide isn't what I want. 

Here's some code to demonstrate what I'm doing since there might be something wrong in my method:

```js
....
const [selectedTile, setSelectedTile] = useState();
....

<Calendar
    {....calendarOptions}
    onBeforeCreateSchedule={(e) => { setSelectedTile(e) }};
/>
```  

## Expected Behavior
What I want to achieve is : 
1. The guide stays where it's been when user clicked the tile
2. the selectedTile state has at least some basic info about the tile, for example, the date it represents
as it's needed later down the road

Thank you in advance!

### PR DESCRIPTION
**Pull request by adhrinae**
### Please check if the PR fulfills these requirements
- [x] It's the right issue type on the title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has a description of the breaking change

### Description

- Migrated to the latest version of the storybook(v6).
- Upgraded the `tui-calendar` version.
- Adopt `react-fast-compare` library to detect actual prop changes.
- Fixed unnecessary re-rendering when the `setOptions` method had been called.
- Updated examples that have possibilities to misguide users.

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,18 +43,18 @@
...
-    "webpack-dev-server": "^3.1.14",
-    "tui-release-notes": "git+https://github.com/nhn/toast-ui.release-notes.git"
+    "tui-release-notes": "git+https://github.com/nhn/toast-ui.release-notes.git",
+    "webpack": "^4.46.0",
+    "webpack-cli": "^3.3.12",
+    "webpack-dev-server": "^3.11.3"
   },
   "dependencies": {
-    "tui-calendar": "^1.12.5"
+    "react-fast-compare": "^3.2.0",
```
Comment (lja1018) [inline package.json:57]: 새 dependencies 추가에 대해 별도의 안내가 없어도 되나요?

**Review by lja1018 — APPROVED** [review]
리뷰 완료합니다.

File: package.json (hunk 1 above)
Comment (adhrinae) [reply package.json:57]: 일단 새로 추가된 문서 내용에 들어있어요. 딱히 zero dependency같은거 표방한 것도 아니라서 괜찮을거에요.

File: .eslintrc.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ module.exports = {
       jsx: true
     }
   },
-  extends: ['tui/es6', 'plugin:react/recommended', 'plugin:prettier/recommended'],
+  extends: ['tui/es6', 'plugin:react/recommended', 'plugin:prettier/recommended', 'plugin:storybook/recommended'],
```
[the remark this answers is not in the document]
Comment (adhrinae) [reply .eslintrc.js:10]: 아예 `sb init` 커맨드로 활성화하니까 기본으로 추가해주더라고요. v2에도 넣어야할듯.

**Review by benkovy — APPROVED** [review]
This looks good to me!! ❤️

**Pull request merged**